### PR TITLE
Commons lib up

### PR DESCRIPTION
**Pull request by EmptyInfinity**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -16,8 +16,8 @@
   "author": "elastic.io GmbH",
   "license": "Apache-2.0",
   "dependencies": {
-    "@elastic.io/component-commons-library": "2.1.0",
-    "@elastic.io/oih-standard-library": "2.0.2",
+    "@elastic.io/component-commons-library": "3.0.0-dev.5",
+    "@elastic.io/oih-standard-library": "2.0.3-dev.1",
```
Comment (if0s) [inline package.json:20]: dev remainder

File: spec/attachments.spec.js [hunk 2]
```diff
@@ -41,11 +49,11 @@ describe('Attachment tests', () => {
       accessTime: '1575379317000',
       modifyTime: '1575291942000',
     };
-    await attachments.addAttachment.call(self, msg, file.name, stream, file.size);
+    await attachments.addAttachment.call(self, msg, file.name, getStream, file.size);
     expect(self.emit.getCall(0).args[1].message).to.be.equal('File size is 70000000000 bytes, it violates the variable MAX_FILE_SIZE, which is currently set to 104857600 bytes');
   });
 
-  it('Emits an error upon failure', async () => {
+  it('Emits an error upon failure', async () => { // ????????
```
Comment (if0s) [inline spec/attachments.spec.js:56]: `// ????????` ?

File: component.json [hunk 3]
```diff
@@ -1,8 +1,6 @@
 {
   "title": "SFTP",
-  "buildType": "docker",
   "description": "Provides file access and transfer using SSH File Transfer Protocol",
-  "version": "1.5.0-dev.2",
```
Comment (if0s) [inline component.json:5]: Get version back :)

File: lib/attachments.js [hunk 4]
```diff
@@ -4,13 +4,13 @@ const { AttachmentProcessor } = require('@elastic.io/component-commons-library')
...
-async function addAttachment(msg, name, stream, contentLength) {
+async function addAttachment(msg, name, getStream, contentLength) {
   try {
     if (contentLength > MAX_FILE_SIZE) {
       throw new Error(`File size is ${contentLength} bytes, it violates the variable MAX_FILE_SIZE, which is currently set to ${MAX_FILE_SIZE} bytes`);
     }
-    const result = await new AttachmentProcessor().uploadAttachment(stream);
-    const attachmentUrl = `${result.config.url}${result.data.objectId}?storage_type=maester`;
+    const attachmentId = await new AttachmentProcessor().uploadAttachment(getStream);
+    const attachmentUrl = new AttachmentProcessor().getMaesterAttachmentUrlById(attachmentId);
```
Comment (if0s) [inline lib/attachments.js:13]:
```suggestion
    const attachmentProcessor = new AttachmentProcessor();
    const attachmentId = await attachmentProcessor.uploadAttachment(getStream);
    const attachmentUrl = attachmentProcessor.getMaesterAttachmentUrlById(attachmentId);
```
Usually better create it once (more memory effective use)

**Review by if0s — APPROVED** [review]

File: spec/attachments.spec.js (hunk 2 above)
Comment (EmptyInfinity) [reply spec/attachments.spec.js:56]: ???

**Pull request merged**